frontend/src/concepts/k8s/K8sNameDescriptionField/K8sNameDescriptionField.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from 'react';
+import {
+  Button,
+  FormGroup,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  Stack,
+  StackItem,
+  TextArea,
+  TextInput,
+} from '@patternfly/react-core';
+import {
+  K8sNameDescriptionFieldData,
+  K8sNameDescriptionFieldUpdateFunction,
+  UseK8sNameDescriptionDataConfiguration,
+  UseK8sNameDescriptionFieldData,
+} from '~/concepts/k8s/K8sNameDescriptionField/types';
+import ResourceNameDefinitionTooltip from '~/concepts/k8s/ResourceNameDefinitionTooltip';
+import useAutoFocusRef from '~/utilities/useAutoFocusRef';
+import { handleUpdateLogic, setupDefaults } from '~/concepts/k8s/K8sNameDescriptionField/utils';
+import ResourceNameField from '~/concepts/k8s/K8sNameDescriptionField/ResourceNameField';
+
+/** Companion data hook */
+export const useK8sNameDescriptionFieldData = (
+  configuration: UseK8sNameDescriptionDataConfiguration = {},
+): UseK8sNameDescriptionFieldData => {
+  const [data, setData] = React.useState<K8sNameDescriptionFieldData>(() =>
+    setupDefaults(configuration),
+  );
+
+  // Hold the data in a ref to avoid churn in the update method
+  const dataRef = React.useRef(data);
+  dataRef.current = data;
+  const onDataChange = React.useCallback<K8sNameDescriptionFieldUpdateFunction>(
+    (key, value) => {
+      setData(handleUpdateLogic(dataRef.current)(key, value));
+    },
+    [setData],
+  );
+
+  return { data, onDataChange };
+};
+
+type K8sNameDescriptionFieldProps = {
+  autoFocusName?: boolean;
+  dataTestId: string;
+  descriptionLabel?: string;
+  nameLabel?: string;
+} & UseK8sNameDescriptionFieldData;
+
+/**
+ * Use in place of any K8s Resource creation / edit.
+ * @see useK8sNameDescriptionFieldData
+ */
+const K8sNameDescriptionField: React.FC<K8sNameDescriptionFieldProps> = ({
+  autoFocusName,
+  data,
+  dataTestId,
+  descriptionLabel = 'Description',
+  onDataChange,
+  nameLabel = 'Name',
+}) => {
+  const autoFocusNameRef = useAutoFocusRef(autoFocusName);
+  const [showK8sField, setShowK8sField] = React.useState(false);
+
+  const { name, description, k8sName } = data;
+
+  return (
+    <Stack hasGutter>
+      <StackItem>
+        <FormGroup label={nameLabel} isRequired>
+          <TextInput
+            data-testid={`${dataTestId}-name`}
+            id={`${dataTestId}-name`}
+            ref={autoFocusNameRef}
```

Comment:
Is there a specific reason this isn't:
```suggestion
            autoFocus={autoFocusName}
```

Reply:
Didn't know PF supported it

Reply:
Oh... they don't? https://www.patternfly.org/components/forms/text-input#textinput

Not sure I follow what you're asking here @christianvogt

Reply:
`autoFocus` is a generic react prop for focusing elements. Not exactly sure if there's any nuance with using it. I don't think it's quite the same as the html `autofocus` attribute but the end result is.

Reply:
TIL.